p2p/src/network.rs

```diff
@@ -317,6 +326,15 @@ where
                 self.broker.issue_send(*msg).await;
             }
         };
+        let mut online_peer_keys = Vec::new();
+        for peer_key in self.peers.keys() {
```

Comment:
```suggestion
        self.peers.keys().iter().cloned().collect();
```
The `iter()` is sometimes redundant, but I like to keep it.

Reply:
yes